presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -213,19 +203,13 @@ public int getEstimatedSizeInBytes()
     {
         int result = INSTANCE_SIZE;
         result += path.length() * Character.BYTES;
-        result += sizeOfObjectArray(partitionKeys.size());
-        for (HivePartitionKey partitionKey : partitionKeys) {
-            result += partitionKey.getEstimatedSizeInBytes();
-        }
         result += sizeOfObjectArray(blocks.size());
         for (InternalHiveBlock block : blocks) {
             result += block.getEstimatedSizeInBytes();
         }
-        result += partitionName.length() * Character.BYTES;
-        result += sizeOfObjectArray(columnCoercions.size());
-        for (HiveTypeName hiveTypeName : columnCoercions.values()) {
-            result += INTEGER_INSTANCE_SIZE + hiveTypeName.getEstimatedSizeInBytes();
-        }
+
+        // partitionInfo is shared between all InternalHiveSplits for a partition
+        result += partitionInfo.getEstimatedSizeInBytes() / partitionInfo.getReferenceCount();
```

Comment:
It's not immediately clear to me how this would work. For example, if `getEstimatedSizeInBytes` are called after *each* `InternalHiveSplit` is discovered, then we probably count the partition info size as:
```
size * (1 + 1/2 + 1/3 + ... + 1/n)
```

When when decrease the size of an `InternalHiveSplit`, we might get a different size in bytes as it was added. Also debugging such memory-tracking issues could be very tricky? 

Instead of amortize the size of `partitionInfo` into each `InternalHiveSplit`, would it be possible to count them independently?

Reply:
good point.  I'll account for it separately